### ISSUE TITLE
multi-line string: `klog.InfoS` prints ASCII escape characters; `klog.Info`, `klog.InfoS` has anomalies

### ISSUE THREAD

**Issue by divmadan** [kubernetes/klog#291]
/kind bug

For a string variable `output`:
```
| Updating installed package 'tap' 
- Getting package install for 'tap' 
/ Getting package metadata for 'tap.tanzu.vmware.com' 
- Updating secret 'tap-tap-install-values' 
- Updating package install for 'tap' 
| Waiting for 'PackageInstall' reconciliation for 'tap' 


Error: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:
  Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)
Error: exit status 1

✖  exit status 1 
```

1. `klog.InfoS` prints ASCII escape characters (`\x1b`) for value string in key-value pairs.

    - `klog.Info("output", output)` prints (desired):
      ```
      | Updating installed package 'tap' 
      - Getting package install for 'tap' 
      / Getting package metadata for 'tap.tanzu.vmware.com' 
      - Updating secret 'tap-tap-install-values' 
      - Updating package install for 'tap' 
      | Waiting for 'PackageInstall' reconciliation for 'tap' 
      
      
      Error: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:
        Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)
      Error: exit status 1
      
      ✖  exit status 1 
      ```

    - `klog.InfoS("", "output", output)` prints: 
      ```
      I0207 16:24:18.962347   92461 tanzu-commands.go:25] "" output="\x1b[K\r| Updating installed package 'tap' \x1b[K\n\x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\n\x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r\\ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\n\x1b[K\r| Updating secret 'tap-tap-install-values' \x1b[K\r/ Updating secret 'tap-tap-install-values' \x1b[K\r- Updating secret 'tap-tap-install-values' \x1b[K\n\x1b[K\r| Updating package install for 'tap' \x1b[K\r/ Updating package install for 'tap' \x1b[K\r- Updating package install for 'tap' \x1b[K\n\x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\n\x1b[K\n\nError: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:\n  Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)\nError: exit status 1\n\n✖  exit status 1 \n"
      ```

2. `klog.Info` is removing other arguments and not printing logging info in case another argument of multi-line string is present. 

    - In the example above, `klog.Info("output", output)` didn't print the argument `"output"`, nor the logging info (flags like time, file, line no. etc.).
    - Similar behaviour is observed in the case of `klog.Infof("output: %s", output)`, where it ignored the format string `"output: %s"` and logging info, and just printed the contents of the variable `output`.

**Comment by pohly** [comment]
I have troubles following what exactly isn't working as expected. Can you provide simple source code snippets that show which invocations don't work as expected and for each show what the expected and actual output is?

**Comment by pohly** [comment]
Ideally you split this up into separate issues.

**Comment by pohly** [comment]
> 1. klog.InfoS prints ASCII escape characters (\x1b) for value string in key-value pairs.

Which version of klog are you using?

> 2. klog.Info is removing other arguments and not printing logging info in case another argument of multi-line string is present.

Your `output`  variable seems to contain various console control characters. I suspect `klog.Info` just dumps those literally into the log output, as requested. Could it be that the logging info is printed but then gets overwritten by the content of `output` because of a carriage return?

Here's an example (from https://unix.stackexchange.com/questions/374592/carriage-return-with-echo-command):
```bash
$ echo -e "This is \r my college"
 my college
```

**Comment by pohly** [comment]
> I was using an older version (v2.30.0) as it was an indirect dependency to another package.

Multi-line output for InfoS was added in v2.40.0.

> Since our code is not public, I cannot share an exact example

A simplified version would have been possible, but that's mute because it looks like `klog.Info` is working as intended.

**Comment by divmadan** [comment]
> Multi-line output for InfoS was added in v2.40.0.

Great, that makes sense.

> A simplified version would have been possible

Sure, for clarification, using the same example:

1. `klog.InfoS`:
    ```go
    package main
    
    import "k8s.io/klog/v2"
    
    func main() {
	    output := "\x1b[K\r| Updating installed package 'tap' \x1b[K\n\x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\n\x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r\\ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\n\x1b[K\r| Updating secret 'tap-tap-install-values' \x1b[K\r/ Updating secret 'tap-tap-install-values' \x1b[K\r- Updating secret 'tap-tap-install-values' \x1b[K\n\x1b[K\r| Updating package install for 'tap' \x1b[K\r/ Updating package install for 'tap' \x1b[K\r- Updating package install for 'tap' \x1b[K\n\x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\n\x1b[K\n\nError: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:\n  Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)\nError: exit status 1\n\n✖  exit status 1 \n"
	    klog.InfoS("using InfoS: ", "output", output)
    }
    ```
    
    `v2.30.0`:
    ```
    ➜ go run a.go
    I0211 10:58:38.257720   13807 a.go:7] "using InfoS: " output="\x1b[K\r| Updating installed package 'tap' \x1b[K\n\x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\n\x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r\\ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\n\x1b[K\r| Updating secret 'tap-tap-install-values' \x1b[K\r/ Updating secret 'tap-tap-install-values' \x1b[K\r- Updating secret 'tap-tap-install-values' \x1b[K\n\x1b[K\r| Updating package install for 'tap' \x1b[K\r/ Updating package install for 'tap' \x1b[K\r- Updating package install for 'tap' \x1b[K\n\x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\n\x1b[K\n\nError: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:\n  Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)\nError: exit status 1\n\n✖  exit status 1 \n"
    ```
    
    `v2.40.1`:
    ```
    ➜ go run a.go            
    I0211 11:01:31.158054   14445 a.go:7] "using InfoS: " output=<
    | Updating installed package 'tap' 
    | Getting package install for 'tap' 
    - Getting package metadata for 'tap.tanzu.vmware.com' 
    - Updating secret 'tap-tap-install-values' 
    - Updating package install for 'tap' 
    | Waiting for 'PackageInstall' reconciliation for 'tap' 
    
    
            Error: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:
              Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)
            Error: exit status 1
    
            ✖  exit status 1 
     >
    ```

2. `klog.Info`:
    ```go
    package main
    
    import "k8s.io/klog/v2"
    
    func main() {
	    output := "\x1b[K\r| Updating installed package 'tap' \x1b[K\n\x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\r/ Getting package install for 'tap' \x1b[K\r- Getting package install for 'tap' \x1b[K\r\\ Getting package install for 'tap' \x1b[K\r| Getting package install for 'tap' \x1b[K\n\x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r\\ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r| Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r/ Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\r- Getting package metadata for 'tap.tanzu.vmware.com' \x1b[K\n\x1b[K\r| Updating secret 'tap-tap-install-values' \x1b[K\r/ Updating secret 'tap-tap-install-values' \x1b[K\r- Updating secret 'tap-tap-install-values' \x1b[K\n\x1b[K\r| Updating package install for 'tap' \x1b[K\r/ Updating package install for 'tap' \x1b[K\r- Updating package install for 'tap' \x1b[K\n\x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r/ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r- Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r\\ Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\r| Waiting for 'PackageInstall' reconciliation for 'tap' \x1b[K\n\x1b[K\n\nError: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:\n  Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)\nError: exit status 1\n\n✖  exit status 1 \n"
	    klog.Info("using Info: ", output)
    }
    ```
    `v2.30.0` and `v2.40.1`:
    ```
    ➜ go run a.go
    | Updating installed package 'tap' 
    | Getting package install for 'tap' 
    - Getting package metadata for 'tap.tanzu.vmware.com' 
    - Updating secret 'tap-tap-install-values' 
    - Updating package install for 'tap' 
    | Waiting for 'PackageInstall' reconciliation for 'tap' 
    
    
    Error: resource reconciliation failed: kapp: Error: waiting on reconcile packageinstall/cnrs (packaging.carvel.dev/v1alpha1) namespace: tap-install:
      Finished unsuccessfully (Reconcile failed:  (message: Error (see .status.usefulErrorMessage for details))). Reconcile failed: Error (see .status.usefulErrorMessage for details)
    Error: exit status 1
    
    ✖  exit status 1 
    ```


***

Final issue resolution:
1. `klog.InfoS` prints ASCII escape characters (\x1b) for value string in key-value pairs.
    > Multi-line output for InfoS was added in v2.40.0.

2. `klog.Info` is removing other arguments and not printing logging info in case another argument of multi-line string is present.
    Logging info gets overwritten because of a carriage return present in the string.
    > Could it be that the logging info is printed but then gets overwritten by the content of output because of a carriage return?
